mathics/core/parser/parser.py

```diff
@@ -236,6 +236,12 @@ def parse_postfix(self, expr1, token, p):
     # Called with one Token and return a Node.
     # Used for prefix operators and brackets.
 
+    def p_PatternTest(self,token):
```

Comment:
Yes, one of the quirks of the parser is that is uses the token name (called `tag` in the code) to construct the Node (see the `parse_p` method). This works well for most cases unless the token is reused across different functions with different names.